### PR TITLE
Explore connectivity options

### PR DESCRIPTION
**Pull request by pipo02mix**
### What this PR does / why we need it
Towards https://github.com/giantswarm/giantswarm/issues/30831

### Things to check/remember before submitting

- If you made content changes

    - Run `make lint dev` to render and proofread content changes locally.
    - Bump `last_review_date` in the front matter header if you reviewed the entire page.

### CODE REVIEW

**Comment by stone-z** [comment]
@pipo02mix are you ready for review (asking because draft status)?

**Comment by pipo02mix** [comment]
NOw it is @stone-z

File: src/content/getting-started/understand-connectivity/_index.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+```sh
+kubectl delete netpol --all -n default
+kubectl apply -f allow-default-namespace.yaml
+$ kubectl exec -it debug-toolbox -- sh
+/ curl -Is hello-world.default
+/ HTTP/1.1 200 OK
+```
+
+We've learnt how network policies work and the peculiarities in Giant Swarm clusters. However, the ideal configuration is to have a default deny policy in all your namespaces and allow only the necessary traffic to your applications. This way, you can control the traffic in your cluster and avoid any security issues.
```
Comment (stone-z) [inline src/content/getting-started/understand-connectivity/_index.md:247]:
```suggestion
This guide has introduced how network policies work and some peculiarities to be aware of when working in Giant Swarm clusters. However, the ideal configuration is to have a default deny policy in all namespaces, and allow only the necessary traffic to and from each application. This approach greatly improves security, and ensures cluster traffic is intentionally and declaratively managed.
```

File: src/content/getting-started/understand-connectivity/_index.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+## Next step
+
+Since you learnt how to configure your application to communicate with other services, now let's dive into how [observe your application]({{< relref "/getting-started/observe-your-app" >}}).
```
Comment (stone-z) [inline src/content/getting-started/understand-connectivity/_index.md:251]:
```suggestion
Now that you've learned how to configure your application to communicate with other services, let's dive into how [observe your application]({{< relref "/getting-started/observe-your-app" >}}).
```

File: src/content/getting-started/understand-connectivity/_index.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+title: Control the application connectivity
+description: Understand how basic connectivity works in the platform and which options are available for exposing your app.
+weight: 60
+last_review_date: 2024-09-01
+menu:
+  principal:
+    parent: getting-started
+    identifier: getting-started-understand-connectivity
+owner:
+  - https://github.com/orgs/giantswarm/teams/sig-cabbage
```
Comment (stone-z) [inline src/content/getting-started/understand-connectivity/_index.md:11]:
```suggestion
  - https://github.com/orgs/giantswarm/teams/team-cabbage
```

File: src/content/getting-started/understand-connectivity/_index.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+Note that the policy is already deployed in the `default` namespace due to the chart enabled by default. It allows the DNS queries to be resolved like:
+
+```sh
+$ kubectl exec debug-toolbox -- nslookup hello-world
+Server:		172.31.0.10
+Address:	172.31.0.10:53
+
+Name:	hello-world.default.svc.cluster.local
+Address: 172.31.104.47
```
Comment (pipo02mix) [inline src/content/getting-started/understand-connectivity/_index.md:168]:
```suggestion
Address: 172.31.104.47
```

File: src/content/getting-started/understand-connectivity/_index.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+__Warning__: By default, clusters run CoreDNS listening on port 1053 (for security reasons). So, you will need to include port `1053` on the list of ports.
+
+Note that the policy is already deployed in the `default` namespace due to the chart enabled by default. It allows the DNS queries to be resolved like:
+
+```sh
+$ kubectl exec debug-toolbox -- nslookup hello-world
+Server:		172.31.0.10
+Address:	172.31.0.10:53
+
+Name:	hello-world.default.svc.cluster.local
```
Comment (pipo02mix) [inline src/content/getting-started/understand-connectivity/_index.md:167]:
```suggestion
Name: hello-world.default.svc.cluster.local
```

File: src/content/getting-started/understand-connectivity/_index.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+```
+
+__Warning__: By default, clusters run CoreDNS listening on port 1053 (for security reasons). So, you will need to include port `1053` on the list of ports.
+
+Note that the policy is already deployed in the `default` namespace due to the chart enabled by default. It allows the DNS queries to be resolved like:
+
+```sh
+$ kubectl exec debug-toolbox -- nslookup hello-world
+Server:		172.31.0.10
+Address:	172.31.0.10:53
```
Comment (pipo02mix) [inline src/content/getting-started/understand-connectivity/_index.md:165]:
```suggestion
Address: 172.31.0.10:53
```

**Comment by pipo02mix** [comment]
@stone-z thanks for the extensive review!

**Review by stone-z — APPROVED** [review]
I'm fine with this but I think you'll need to fix the Vale linting suggestions. Seems like the build itself is also failing but I think it's related to this page not existing yet

File: src/content/getting-started/understand-connectivity/_index.md [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+ - from:
+ - namespaceSelector:
+          matchLabels:
+            name: default
+```
+
+Note that the namespace to which this policy applies needs to carry a label `name:` similar to the actual name key in its metadata:
+
+```sh
+$ kubectl label namespace default name=default
```
Comment (pipo02mix) [inline src/content/getting-started/understand-connectivity/_index.md:237]:
```suggestion
kubectl label namespace default name=default
```

File: src/content/getting-started/understand-connectivity/_index.md [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+
+Note that the policy is already deployed in the `default` namespace due to the chart enabled by default. It allows the DNS queries to be resolved like:
+
+```sh
+$ kubectl exec debug-toolbox -- nslookup hello-world
+Server:		172.31.0.10
+Address: 172.31.0.10:53
+
+Name: hello-world.default.svc.cluster.local
+Address: 172.31.104.47
```
Comment (pipo02mix) [inline src/content/getting-started/understand-connectivity/_index.md:169]:
```suggestion
kubectl exec debug-toolbox -- nslookup hello-world
Server: 172.31.0.10
Address: 172.31.0.10:53

Name: hello-world.default.svc.cluster.local
Address: 172.31.104.47
```

**Pull request merged**